core-bundle/contao/drivers/DC_Table.php

```diff
@@ -4021,20 +4021,25 @@ protected function treeView()
 
 		$topMostRootIds = $this->root;
 
-		if (!empty($this->visibleRootTrails) || !empty($this->root))
+		if (isset($GLOBALS['TL_DCA'][$table]['list']['sorting']['visibleRoot']))
 		{
-			if (isset($GLOBALS['TL_DCA']['tl_page']['list']['sorting']['visibleRoot']))
-			{
-				$topMostRootIds = array($GLOBALS['TL_DCA']['tl_page']['list']['sorting']['visibleRoot']);
-			}
-			else
-			{
-				// Make sure we use the topmost root IDs only from all the visible root trail ids and also ensure correct sorting
-				$topMostRootIds = $db
-					->prepare("SELECT id FROM $table WHERE (pid=0 OR pid IS NULL) AND id IN (" . implode(',', array_merge($this->visibleRootTrails, $this->root)) . ")" . ($db->fieldExists('sorting', $table) ? ' ORDER BY sorting, id' : ''))
-					->execute()
-					->fetchEach('id');
-			}
+			$topMostRootIds = array($GLOBALS['TL_DCA'][$table]['list']['sorting']['visibleRoot']);
```

Comment:
This was previously only set if `!empty($this->visibleRootTrails) || !empty($this->root)` was true. Are you sure that changing this does not have side effects?

Reply:
So there will be side effects 😄